### PR TITLE
feat(nuttx): update the joined invalid area instead of the last one

### PR DESCRIPTION
**Pull request by jianglianfang**
### Description of the feature or fix

update the joined invalid area instead of the last one. Otherwise it will cause update area flashing when FB_UPDATE enbled.

### Notes
- Update the [Documentation](https://github.com/lvgl/lvgl/tree/master/docs) if needed.
- Add [Examples](https://github.com/lvgl/lvgl/tree/master/examples) if relevant.
- Add [Tests](https://github.com/lvgl/lvgl/blob/master/tests/README.md) if applicable.
- If you added new options to `lv_conf_template.h` run [lv_conf_internal_gen.py](https://github.com/lvgl/lvgl/blob/master/scripts/lv_conf_internal_gen.py) and update [Kconfig](https://github.com/lvgl/lvgl/blob/master/Kconfig).
- Run `scripts/code-format.py` ([astyle](http://astyle.sourceforge.net/install.html) version [v3.4.12](https://github.com/szepeviktor/astyle/releases/tag/v3.4.12) needs to be installed) and follow the [Code Conventions](https://docs.lvgl.io/master/CODING_STYLE.html).
- Mark the Pull request as [Draft](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/changing-the-stage-of-a-pull-request) while you are working on the first version, and mark is as _Ready_ when it's ready for review.
- When changes were requested, [re-request review](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/requesting-a-pull-request-review) to notify the maintainers.
- Help us to review this Pull Request! Anyone can [approve or request changes](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/approving-a-pull-request-with-required-reviews).

### CODE REVIEW

File: src/display/lv_display.c [hunk 1]
```diff
@@ -933,6 +933,36 @@ lv_draw_buf_t * lv_display_get_buf_active(lv_display_t * disp)
     return disp->buf_act;
 }
 
+bool lv_display_get_dirty_area(lv_display_t * disp, lv_area_t * dirty_area)
```
Comment (FASTSHIFT) [inline src/display/lv_display.c]: How about `lv_display_get_invalid_area` ?

**Comment by kisvegabor** [comment]
Can we add an `lv_area_t dirty_area` element to `lv_nuttx_fb_t` where you can join all areas before
```c
 if(!lv_display_flush_is_last(disp)) {
        lv_display_flush_ready(disp);
        return;
    }
```
The area can be reset after  the `ioctl`.

It would save us adding a new API function and give more flexibility.

File: src/display/lv_display.c (hunk 1 above)
Comment (jianglianfang) [reply src/display/lv_display.c]: move to lv_nuttx

File: src/drivers/nuttx/lv_nuttx_fbdev.c [hunk 2]
```diff
@@ -169,6 +169,34 @@ int lv_nuttx_fbdev_set_file(lv_display_t * disp, const char * file)
...
+#if defined(CONFIG_FB_UPDATE)
+static void fbdev_join_inv_areas(lv_display_t * disp, lv_area_t * final_inv_area)
+{
+    uint16_t inv_index;
+
+    bool area_joined = false;
+
+    for(inv_index = 0; inv_index < disp->inv_p; inv_index++) {
+        if(disp->inv_area_joined[inv_index] == 0) {
+            FAR const lv_area_t * area_p = &disp->inv_areas[inv_index];
```
Comment (FASTSHIFT) [inline src/drivers/nuttx/lv_nuttx_fbdev.c]: No need `FAR`.

File: src/drivers/nuttx/lv_nuttx_fbdev.c [hunk 3]
```diff
@@ -169,6 +169,34 @@ int lv_nuttx_fbdev_set_file(lv_display_t * disp, const char * file)
  *   STATIC FUNCTIONS
  **********************/
 
+#if defined(CONFIG_FB_UPDATE)
+static void fbdev_join_inv_areas(lv_display_t * disp, lv_area_t * final_inv_area)
```
Comment (liamHowatt) [inline src/drivers/nuttx/lv_nuttx_fbdev.c:176]: Please forward-declare it in the static prototypes section

**Review by liamHowatt — COMMENTED** [review]
Makes sense to me!

File: src/drivers/nuttx/lv_nuttx_fbdev.c (hunk 3 above)
Comment (jianglianfang) [reply src/drivers/nuttx/lv_nuttx_fbdev.c:176]: done

File: src/drivers/nuttx/lv_nuttx_fbdev.c (hunk 2 above)
Comment (jianglianfang) [reply src/drivers/nuttx/lv_nuttx_fbdev.c]: done

**Review by liamHowatt — APPROVED** [review]
Thanks.

I assume `disp->inv_area_joined[inv_index]` will be `1` for at least one value of `inv_index`. From what I could discern in lv_refr.c, that seems to be the case.

**Review by FASTSHIFT — APPROVED** [review]

**Pull request merged**